src/python/grpcio/grpc/beta/implementations.py

```diff
@@ -73,12 +73,75 @@ def ssl_client_credentials(root_certificates, private_key, certificate_chain):
       certificate chain should be used.
 
   Returns:
-    A ClientCredentials for use with an SSL-enabled Channel.
+    A ChannelCredentials for use with an SSL-enabled Channel.
   """
-  intermediary_low_credentials = _intermediary_low.ClientCredentials(
-      root_certificates, private_key, certificate_chain)
-  return ClientCredentials(
-      intermediary_low_credentials._internal, intermediary_low_credentials)  # pylint: disable=protected-access
+  return ChannelCredentials(_low.channel_credentials_ssl(
+      root_certificates, private_key, certificate_chain))
+
+
+class CallCredentials(object):
+  """A value encapsulating data asserting an identity over an *established*
+  channel. May be composed with ChannelCredentials to always assert identity for
+  every call over that channel.
+
+  This class and its instances have no supported interface - it exists to define
+  the type of its instances and its instances exist to be passed to other
+  functions.
+  """
+
+  def __init__(self, low_credentials):
+    self._low_credentials = low_credentials
+
+
+def metadata_call_credentials(metadata_callback, name=None):
+  """Construct call credentials from an auth metadata plugin.
```

Comment:
> The return type itself is in the return type doc-comment below, and I'd feel a bit weird then calling the auth metadata plugin an 'auth metadata plugin' as opposed to giving it some proper type.

I'm OK with that but then, I think it would make sense to change the doc for the ssl_credentials which uses `ChannelCredentials` as opposed to `channel credentials` (see my new comment on this).

> Actually, that's a question: should the metadata_callback argument actually be some metadata_auth_plugin argument with an abstract class interface somewhere? I'm leaning towards no, for ease of use, but just bringing it up as a possibility.

This more of a python question I believe. What would you, user of the API, think would be the most appropriate? Maybe @nathanielmanistaatgoogle has some good insight on this.